src/writer.rs

```diff
@@ -394,4 +458,75 @@ mod indentation {
                 .as_ref()
         );
     }
+    #[test]
+    fn element_writer_empty() {
+        let mut buffer = Vec::new();
+        let mut writer = Writer::new_with_indent(&mut buffer, b' ', 4);
+
+        writer
+            .create_element(b"empty")
+            .with_attribute(("attr1", "value1"))
+            .with_attribute(("attr2", "value2"))
+            .write_empty()
+            .expect("failure");
+
+        assert_eq!(
+            std::str::from_utf8(&buffer).unwrap(),
+            r#"<empty attr1="value1" attr2="value2"/>"#
+        );
+    }
+
+    #[test]
+    fn element_writer_text() {
+        let mut buffer = Vec::new();
+        let mut writer = Writer::new_with_indent(&mut buffer, b' ', 4);
+
+        writer
+            .create_element(b"paired")
+            .with_attribute(("attr1", "value1"))
+            .with_attribute(("attr2", "value2"))
+            .write_text_content(BytesText::from_plain_str("text"))
+            .expect("failure");
+
+        assert_eq!(
+            std::str::from_utf8(&buffer).unwrap(),
+            r#"<paired attr1="value1" attr2="value2">text</paired>"#
+        );
+    }
+
+    #[test]
+    fn element_writer_scope() {
```

Comment:
I wanted to try this experiment, but I'm not strongly attached to it. What do you think?  It seems useful for really simple cases of nesting, but less so more complex ones.